src/main/java/net/masterthought/cucumber/json/Hook.java

```diff
@@ -38,4 +39,21 @@ public Match getMatch() {
     public Embedding[] getEmbeddings() {
         return embeddings;
     }
+
+    /**
+     * Checks if the hook has content meaning as it has at least attachment or result with error message.
+     *
+     * @return <code>true</code> if the hook has content otherwise <code>false</code>
+     */
+    public boolean hasContent() {
+        if (embeddings.length > 0) {
+            // assuming that if the embedding exists then it is not empty
+            return true;
+        }
+        if (StringUtils.isNotBlank(result.getErrorMessage())) {
+            return true;
+        }
+        // TODO: hook with 'output' should be treated as empty or not?
```

Comment:
IMO this implementation is OK. I have not seen important output from hooks (aside from embeddings / exception error messages) yet.
If anyone should miss further output from hooks, she can simply disable the hide_empty_hooks feature.
So as far as I am concerned, feel free to remove TODO comment.